src/services/bundleAnalysis/useBundleSummary.tsx

```diff
@@ -0,0 +1,190 @@
+import A from 'ui/A'
+
+const BundleDataSchema = z.object({
+  loadTime: z.object({
+    threeG: z.number(),
+    highSpeed: z.number(),
+  }),
+  size: z.object({
+    gzip: z.number(),
+    uncompress: z.number(),
+  }),
+})
+
+const BundleAnalysisReportSchema = z.object({
+  __typename: z.literal('BundleAnalysisReport'),
+  bundle: z
+    .object({
+      name: z.string(),
+      moduleCount: z.number(),
+      sizeTotal: z.number(),
+      bundleData: BundleDataSchema,
+    })
+    .nullable(),
+})
+
+const BundleReportSchema = z.discriminatedUnion('__typename', [
+  BundleAnalysisReportSchema,
+  MissingHeadReportSchema,
+])
+
+const RepositorySchema = z.object({
+  __typename: z.literal('Repository'),
+  branch: z
+    .object({
+      head: z
+        .object({
+          bundleAnalysisReport: BundleReportSchema.nullable(),
+        })
+        .nullable(),
+    })
+    .nullable(),
+})
+
+const RequestSchema = z.object({
+  owner: z
+    .object({
+      repository: z
+        .discriminatedUnion('__typename', [
+          RepositorySchema,
+          RepoNotFoundErrorSchema,
+          RepoOwnerNotActivatedErrorSchema,
+        ])
+        .nullable(),
+    })
+    .nullable(),
+})
+
+const query = `
+query BundleSummary(
+  $owner: String!
+  $repo: String!
+  $branch: String!
+  $bundle: String!
+) {
+  owner(username: $owner) {
+    repository(name: $repo) {
+      __typename
+      ... on Repository {
+        branch(name: $branch) {
+          head {
+            bundleAnalysisReport {
+              __typename
+              ... on BundleAnalysisReport {
+                bundle(name: $bundle) {
+                  name
+                  moduleCount
+                  sizeTotal
+                  bundleData {
+                    loadTime {
+                      threeG
+                      highSpeed
+                    }
+                    size {
+                      gzip
+                      uncompress
+                    }
+                  }
+                }
+              }
+              ... on MissingHeadReport {
+                message
+              }
+            }
+          }
+        }
+      }
+      ... on NotFoundError {
+        message
+      }
+      ... on OwnerNotActivatedError {
+        message
+      }
+    }
+  }
+}`
+
+interface UseBundleSummaryArgs {
+  provider: string
+  owner: string
+  repo: string
+  branch: string
+  bundle: string
+}
+
+export const useBundleSummary = ({
+  provider,
+  owner,
+  repo,
+  branch,
+  bundle,
+}: UseBundleSummaryArgs) => {
+  return useQuery({
+    queryKey: ['BundleSummary', provider, owner, repo, branch, bundle],
+    queryFn: ({ signal }) =>
+      Api.graphql({
+        provider,
+        query,
+        signal,
+        variables: { owner, repo, branch, bundle },
+      }).then((res) => {
+        const parsedData = RequestSchema.safeParse(res?.data)
+
+        if (!parsedData.success) {
+          return Promise.reject({
+            status: 404,
```

Comment:
nit: would this be better as a 400? Signaling a bad request instead of not found

Reply:
Hrrm we're pretty standard across the board right now setting this as 404, if we're going to change to 400, we should also move all of the other ones at the same time. Are you able to create a ticket to address this?